packages/@aws-cdk/aws-lambda/lib/adot-layers.ts

```diff
@@ -0,0 +1,240 @@
+export class AdotLambdaLayerJavaSdkVersion extends AdotLambdaLayerVersionBase {
+  /**
+   * Will always use the latest layer version. New versions could
+   * introduce incompatible changes. Make sure to test new versions.
+   */
+  public static readonly LATEST = new AdotLambdaLayerJavaSdkVersion('1.19.0');
+
+  /**
+   * Version 1.19.0
+   */
+  public static readonly V1_19_0 = new AdotLambdaLayerJavaSdkVersion('1.19.0');
+}
+
+/**
+ * The collection of versions of the ADOT Lambda Layer for Java auto-instrumentation
+ */
+export class AdotLambdaLayerJavaAutoInstrumentationVersion extends AdotLambdaLayerVersionBase {
+  /**
+   * Will always use the latest layer version. New versions could
+   * introduce incompatible changes. Make sure to test new versions.
+   */
+  public static readonly LATEST = new AdotLambdaLayerJavaAutoInstrumentationVersion('1.19.2');
+
+  /**
+   * Version 1.19.2
+   */
+  public static readonly V1_19_2 = new AdotLambdaLayerJavaAutoInstrumentationVersion('1.19.2');
+}
+
+/**
+ * The collection of versions of the ADOT Lambda Layer for Python SDK
+ */
+export class AdotLambdaLayerPythonSdkVersion extends AdotLambdaLayerVersionBase {
+  /**
+   * Will always use the latest layer version. New versions could
+   * introduce incompatible changes. Make sure to test new versions.
+   */
+  public static readonly LATEST = new AdotLambdaLayerPythonSdkVersion('1.13.0');
+
+  /**
+   * Version 1.13.0
+   */
+  public static readonly V1_13_0 = new AdotLambdaLayerPythonSdkVersion('1.13.0');
+}
+
+/**
+ * The collection of versions of the ADOT Lambda Layer for JavaScript SDK
+ */
+export class AdotLambdaLayerJavaScriptSdkVersion extends AdotLambdaLayerVersionBase {
+  /**
+   * Will always use the latest layer version. New versions could
+   * introduce incompatible changes. Make sure to test new versions.
+   */
+  public static readonly LATEST = new AdotLambdaLayerJavaScriptSdkVersion('1.7.0');
+
+  /**
+   * Version 1.7.0
+   */
+  public static readonly V1_7_0 = new AdotLambdaLayerJavaScriptSdkVersion('1.7.0');
+}
+
+/**
+ * The collection of versions of the ADOT Lambda Layer for generic purpose
+ */
+export class AdotLambdaLayerGenericVersion extends AdotLambdaLayerVersionBase {
+  /**
+   * Will always use the latest layer version. New versions could
+   * introduce incompatible changes. Make sure to test new versions.
+   */
+  public static readonly LATEST = new AdotLambdaLayerGenericVersion('0.62.1');
+
+  /**
+   * Version 0.62.1
+   */
+  public static readonly V0_62_1 = new AdotLambdaLayerGenericVersion('0.62.1');
+}
+
+/**
+ * Properties for an ADOT instrumentation in Lambda
+ */
+export interface AdotInstrumentationProps {
+  /**
+   * The type of the ADOT Lambda layer.
+   */
+  readonly type: AdotLambdaLayerType;
+
+  /**
+   * The startup script to run, see ADOT documentation to pick the right script for your use case: https://aws-otel.github.io/docs/getting-started/lambda
+   */
+  readonly execWrapper: string;
```

Comment:
Can we make this an enum? Looking at the docs it looks like there are only a
couple of valid options.

```ts
class AdotExecWrapper {
  public static HANDLER = '/opt/otel-handler',
  public static PROXY_HANDLER = '/opt/otel-proxy-handler',
  public static STREAM_HANDLER = '/opt/otel-stream-handler',

  public static customHandler(handler: string): string {
    return handler;
  }
}
```

Reply:
We can. I've updated this in the latest revision. I chose to use a real enum instead of a enum-like class to make the interface simpler. If users really want to use a custom handler, they can always override the `AWS_LAMBDA_EXEC_WRAPPER` environment variable themselves with `fn.addEnvironment();`.

This is a two-way door as we can still switch to the enum-like class option later (if necessary).